pkg/reconciler/configuration/configuration.go

```diff
@@ -196,64 +212,118 @@ func (c *Reconciler) reconcile(ctx context.Context, config *v1alpha1.Configurati
 		return err
 	}
 
+	if (rc == nil || rc.Status != corev1.ConditionTrue) && sortedRevisions != nil {
+		// Update the LatestReadyRevisionName to the last ready revision
+		for _, rev := range sortedRevisions {
+			if rev.Status.IsReady() {
+				config.Status.SetLatestReadyRevisionName(rev.Name)
+				break
+			}
+		}
+	}
+
 	return nil
 }
 
 // CheckNameAvailability checks that if the named Revision specified by the Configuration
 // is available (not found), exists (but matches), or exists with conflict (doesn't match).
-func CheckNameAvailability(config *v1alpha1.Configuration, lister listers.RevisionLister) (*v1alpha1.Revision, error) {
+func CheckNameAvailability(config *v1alpha1.Configuration, lister listers.RevisionLister) error {
 	// If config.Spec.GetTemplate().Name is set, then we can directly look up
 	// the revision by name.
 	name := config.Spec.GetTemplate().Name
 	if name == "" {
-		return nil, nil
+		return nil
 	}
 	errConflict := errors.NewAlreadyExists(v1alpha1.Resource("revisions"), name)
 
 	rev, err := lister.Revisions(config.Namespace).Get(name)
 	if errors.IsNotFound(err) {
 		// Does not exist, we must be good!
 		// note: for the name to change the generation must change.
-		return nil, err
+		return err
 	} else if err != nil {
-		return nil, err
+		return err
 	} else if !metav1.IsControlledBy(rev, config) {
 		// If the revision isn't controller by this configuration, then
 		// do not use it.
-		return nil, errConflict
+		return errConflict
 	}
 
 	// Check the generation on this revision.
 	generationKey := serving.ConfigurationGenerationLabelKey
 	expectedValue := resources.RevisionLabelValueForKey(generationKey, config)
 	if rev.Labels != nil && rev.Labels[generationKey] == expectedValue {
-		return rev, nil
+		return nil
 	}
 	// We only require spec equality because the rest is immutable and the user may have
 	// annotated or labeled the Revision (beyond what the Configuration might have).
 	if !equality.Semantic.DeepEqual(config.Spec.GetTemplate().Spec, rev.Spec) {
-		return nil, errConflict
+		return errConflict
 	}
-	return rev, nil
+	return nil
 }
 
-func (c *Reconciler) latestCreatedRevision(config *v1alpha1.Configuration) (*v1alpha1.Revision, error) {
-	if rev, err := CheckNameAvailability(config, c.revisionLister); rev != nil || err != nil {
-		return rev, err
+func (c *Reconciler) getSortedCreatedRevisions(ctx context.Context, config *v1alpha1.Configuration) ([]*v1alpha1.Revision, error) {
+	logger := logging.FromContext(ctx)
+	if err := CheckNameAvailability(config, c.revisionLister); err != nil {
+		return nil, err
```

Comment:
This still changed, and breaks BYO revision name since line 243 will return "Not Found".

Reply:
@mattmoor I'm not getting the "Not found" error with the rollback e2e test https://github.com/knative/serving/pull/5702 (because `checkNameAvailability(...)` is able to find a Configuration with the template name). Could you confirm if I'm doing the rollback test correctly?